daml/Marketplace/Trading/Model.daml

```diff
@@ -73,9 +79,55 @@ template Order
     providerOrderId : Optional Text
     executions : [Execution]
     remainingQuantity : Decimal
-    depositCid : ContractId AssetDeposit
+    collateral : TradeCollateral
   where
     signatory operator, provider, customer
+    observer case collateral of
+      (Cleared ch) -> [ch]
+      _            -> []
 
     key (provider, details.id.label) : (Party, Text)
     maintainer key._1
+
+template ClearedTrade
+  with
+    operator : Party
+    provider : Party
+    clearinghouse : Party
+    makerOrder : Order
+    takerOrder : Order
+    execution : Execution
+  where
+    signatory provider
+    observer clearinghouse
+
+    controller clearinghouse can
+      ClearedTrade_Novate : (ContractId ClearedTradeSide, ContractId ClearedTradeSide)
+        do
+          timeNovated <- getTime
+
+          makerCid <- create ClearedTradeSide with
+            exchange = provider
+            order = makerOrder
+            participant = makerOrder.customer
+            ..
+
+          takerCid <- create ClearedTradeSide with
+            exchange = provider
+            order = takerOrder
+            participant = takerOrder.customer
+            ..
+
+          return (makerCid, takerCid)
+
+template ClearedTradeSide
```

Comment:
Whats the purpose of this `ClearedTradeSide` ?

Reply:
This is so the clearinghouse and the participant has a record that the trade was novated and is the representation of the contract between the participant and the clearinghouse (as central counterparty) for this particular trade. In practice this is what baymarkets is using to keep track of trades to perform mtm/margin adjustments. This was ported over from the old model, so I'm open to different ways of handling this.

Reply:
Would the performing of mtm/margin adjustments be performed by another service ?

Reply:
Hmmm...the actual transfers related to the adjustments are done by `Clearing.Service`, are you suggesting a separate service might be responsible for doing the calculations?

Reply:
I was just wondering who was acting on this template. Maybe it would be clearer to put it in a `Clearing.Model` and call it `ClearingInstruction` or something like that ?

Reply:
I also think that this is leaking Baymarkets specifics into this module. 

It's mixing trading concerns with clearing concerns. I think you should endeavour to keep them separate.

Could the concept of novation be made more general?

Reply:
Just to clarify, `ClearedTradeSide` was a v1 marketplace concept that Baymarkets was using, not necessarily a Baymarkets specific data structure that we put in for them. I do agree though that it may make more sense renamed in the clearing module, since at this point it isn't something the trading service cares about or needs to keep track of.

Reply:
Novation could definitely be more generalized, although I think it's out of the scope of this PR.

Reply:
To the comment: "I would think about how to separate the clearing and settlement concerns such as margin and collateral from the order and trade economics." this may need a larger restructuring since the collateral as a `ContractId AssetDeposit` was already integrated into the `Trading` service. Something else that may make sense to refactor in another PR.